### PR TITLE
executor, planner: support split table regions on clustered index

### PR DESCRIPTION
**Pull request by tangenta**
### What problem does this PR solve?

Problem Summary: 
`SPLIT TABLE` only supports specifying exactly one value as the split point:
```sql
set @@tidb_enable_clustered_index=1;
create table t (a int, b int, c varchar(255), primary key (b, c));
split table t between (0, 'aa') and (0, 'zz') regions 10;
```
```console
(1105, 'Split table region lower value count should be 1')
```

### What is changed and how it works?

What's Changed:

`split table` now can be used on the clustered index tables:

```sql
set @@tidb_enable_clustered_index=1;
create table t (a int, b int, c varchar(255), primary key (b, c));
split table t between (0, 'aa') and (0, 'zz') regions 10;
split table t by (10, 'a'), (100, 'a'), (500, 'a');
```
```console
+--------------------+----------------------+
| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
+--------------------+----------------------+
| 9                  | 1.0                  |
+--------------------+----------------------+
+--------------------+----------------------+
| TOTAL_SPLIT_REGION | SCATTER_FINISH_RATIO |
+--------------------+----------------------+
| 3                  | 1.0                  |
+--------------------+----------------------+

```

How it Works:

### Check List 

Tests 

- Unit test
- Integration test

Side effects

### Release note 

- No release note

### CODE REVIEW

File: planner/core/handle_cols.go [hunk 1]
```diff
@@ -219,3 +219,7 @@ func (ib *IntHandleCols) Compare(a, b []types.Datum) (int, error) {
 	}
 	return 1, nil
 }
+
```
[the remark this answers is not in the document]
Comment (coocood) [reply planner/core/handle_cols.go:222]: reset LGTM

**Comment by coocood** [comment]
LGTM

File: executor/split.go [hunk 2]
```diff
@@ -548,19 +565,49 @@ func (e *SplitTableRegionExec) calculateBoundValue() (lowerValue int64, step int
 	return lowerValue, step, nil
 }
 
-func (e *SplitTableRegionExec) getSplitTablePhysicalKeysFromBound(physicalID, low, step int64, keys [][]byte) [][]byte {
+func (e *SplitTableRegionExec) getSplitTablePhysicalKeysFromBound(physicalID int64, keys [][]byte) ([][]byte, error) {
 	recordPrefix := tablecodec.GenTableRecordPrefix(physicalID)
 	// Split a separate region for index.
-	if len(e.tableInfo.Indices) > 0 {
+
```
Comment (wjhuang2016) [inline executor/split.go:571]: Remove the empty line.

File: planner/core/planbuilder.go [hunk 3]
```diff
@@ -2671,35 +2666,25 @@ func (b *PlanBuilder) buildSplitTableRegion(node *ast.SplitRegionStmt) (Plan, er
 	if len(node.SplitOpt.ValueLists) > 0 {
 		values := make([][]types.Datum, 0, len(node.SplitOpt.ValueLists))
 		for i, valuesItem := range node.SplitOpt.ValueLists {
-			if len(valuesItem) > 1 {
-				return nil, ErrWrongValueCountOnRow.GenWithStackByArgs(i + 1)
-			}
-			value, err := b.convertValue(valuesItem[0], mockTablePlan, pkCol)
+			data, err := convertValueListToData(valuesItem, handleColInfos, i, convertFn)
```
Comment (wjhuang2016) [inline planner/core/planbuilder.go:2669]: Can we not use convertFn as an argument?

**Pull request merged**